### PR TITLE
Fix example extensions and move them to `create-extension` script

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes # .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Load this branch and run `pnpm i` and then `pnpm run create-extension` and run `pnpm i && pnpm run start` on the newly created directory, this should all work correctly.
2. You may also want to add the plugin to a Wordpress site to see if it doesn't break anything.
3. Now run `pnpm run create-extension --example <example name>` -> `pnpm run create-extension --example add-task` for each example listed in the [README](https://github.com/woocommerce/woocommerce/blob/5bc6520e7b43a85a5c3bd2362891d775f289672d/tools/create-extension/README.md).
4. Run `pnpm i && pnpm run build` for each generated example and enable the plugin in Wordpress and make sure it works correctly.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: tools/create-extension/README.md [hunk 1]
```diff
@@ -20,3 +20,20 @@ The script will create a sibling directory by a name of your choosing. Once you
...
+#### Example Extensions
+
+-   `add-report` - Create a "Hello World" report page.
+-   `add-task` - Create a custom task for the onboarding task list.
+-   `dashboard-section` - Adding a custom "section" to the new dashboard area.
+-   `table-column` - An example of how to add column(s) to any report.
+-   `sql-modification` - An example of how to modify SQL statements.
+-   `add-abbreviated-notification` - An example of how to add a abbreviated notification under the Activity panel.
+-   `payment-gateway-suggestions` - An example of how to add custom payment gateways to the WooCommerce task list.
+-   `simple-inbox-note` - An example of how to add a inbox note to the WooCommerce inbox.
```
Comment (psealock) [inline tools/create-extension/README.md:39]: Ok, this is amazing. I was just talking about having something like this

File: tools/create-extension/_package.json [hunk 2]
```diff
@@ -20,11 +20,11 @@
...
-		"@wordpress/scripts": "^12.2.1",
-		"@woocommerce/eslint-plugin": "1.1.0",
-		"@woocommerce/dependency-extraction-webpack-plugin": "1.6.0"
+		"@wordpress/scripts": "^25.1.0",
+		"@woocommerce/eslint-plugin": "2.2.0",
+		"@woocommerce/dependency-extraction-webpack-plugin": "2.2.0"
 	},
 	"dependencies": {
-		"@wordpress/hooks": "^3.5.0"
+		"@wordpress/hooks": "^3.24.0"
```
Comment (psealock) [inline tools/create-extension/_package.json:28]: Moving this tool to `@woocommerce/create-woo-extension` solves the problem of having to keep up dependencies like this.

**Review by psealock — COMMENTED** [review]
This is really fantastic @louwie17 ! 

As I mentioned, this tool is being deprecated in favour of [`@woocommerce/create-woo-extension`](https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/create-woo-extension) which does the same function but can be run from anywhere, placing the resulting repo where the user wants. It also avoids having to touch the Monorepo altogether, which is a bonus for someone just looking to poke around. It uses `@wordpress/create-block` under the hood.

I see you've introduced Commander to handle arguments. We don't have that in the new tool, but are contemplating creating a wrapper to handle this scenario and simplifying the verbose command. See [this conversation](https://github.com/woocommerce/platform-private/issues/126#issuecomment-1347633374). 

Interestingly enough, I was just having a conversation with @shaybanshee about adding functionality similar to what you've done here. Currently, `@wordpress/create-block` only accepts a static `module.exports` in an `index.js` configuration file. I'm trying t[o amend the behaviour to allow for async requests](https://github.com/WordPress/gutenberg/pull/46456), but maybe this will work with Commander as is.

We would change the command to something like this: 

```
npx @woocommerce/create-woo-extension my-extension --add-report
```
Does this make sense? Its a bit of a ramble, but if you have a look at the project maybe we can come up with a plan to integrate this.

**Comment by louwie17** [comment]
Thanks for having a look at this @psealock 

> Does this make sense? Its a bit of a ramble, but if you have a look at the project maybe we can come up with a plan to integrate this.

This does indeed make sense. I did have a relatively detailed look at the `create-block` package and I wondering if we could actually make use of `variants`?
As it stands they are not insanely useful for our examples, but I wonder if we just update [this function/logic](https://github.com/WordPress/gutenberg/blob/trunk/packages/create-block/lib/templates.js#L69) to also make use of the `variant` `outputTemplatesPath` and merge these two together that we can easily port over our current logic.

Although for now we could still do a slightly hackier solution where we make use of `variants` still to set the specified npm dependencies and such, then at the start of the `index.js` file we could do something like this:
```
const variantIndex = process.argv.indexOf( '--variant' );
let templatesPath = __dirname;
if ( variantIndex > -1 && process.argv[ variantIndex + 1 ] ) {
	templatesPath = __dirname + '/examples/' + process.argv[ variantIndex + 1 ];
}
```
The only downside of this, is that we will have to re-use all the `.mustache` files and copy those within each `example` directory. Let me push up the quick example, see this commit: https://github.com/woocommerce/woocommerce/pull/36489/commits/187b4a8ef0df7fa5c34635d9cef6e6075f3ff91a 

I could generate the example by running: `npx @wordpress/create-block -t woocommerce-monorepo/packages/js/create-woo-extension --variant add-task test`

**Comment by psealock** [comment]
@louwie17 Its becoming clear that we'll want to have more control over the output than create-block is giving us. Not only will we want to dynamically pull in version numbers for WooCommerce Core, but we will make use of interactivity for asking the user about options they'd like to add, similar to what you're attempting to do here.

This means the invoking command will change to this:

```
npx @woocommerce/create-woo-extension my-extension-name
```
And `@wordpress/create-block` would be called under the hood. Of course, this then begs the question - would we need `@wordpress/create-block` at all at this point? I think that question can be asked later

**Comment by louwie17** [comment]
Sounds good with me @psealock!

What should the next steps for this PR be then? Should I move over all the extension examples similar to what I started here: https://github.com/woocommerce/woocommerce/pull/36489/commits/187b4a8ef0df7fa5c34635d9cef6e6075f3ff91a 
Or should I wait?
I be happy to help, from our perspective it would be nice to get the extension stuff moved over on the quicker side as part of our extensibility work with the new product screen.

**Comment by psealock** [comment]
>What should the next steps for this PR be then?

The first step would be this https://github.com/woocommerce/woocommerce/issues/36575

Then you could figure out a way to fold in the logic from this PR

**Comment by samueljseay** [comment]
@louwie17 Hey did you want to follow up on this? No pressure :)

**Comment by louwie17** [comment]
> The first step would be this #36575

My plan was to wait on this to be finished: https://github.com/woocommerce/woocommerce/issues/36575 (the one @psealock referred to). 
Unless you have another suggestion? As this did work as is.
Our need for this has delayed a little bit, so we aren't in as a crazy hurry anymore as prior.

**Comment by samueljseay** [comment]
@louwie17 no probs. I'll close this for now until we get that one solved.

**Comment by louwie17** [comment]
Sounds good, thanks @samueljseay

**Pull request closed without merging**